content/2022/prepoznavanje-govora.md

```diff
@@ -2,3 +2,239 @@
+
+Rezultati su prikazani u tabeli ispod.
+
+![Rezultati](static\images\4.png)
+
+Metrika ovih rezultata bila je tačnost, zato što je, zbog izbalansirane baza, ovo reprezentativna metrika.
+
+Rezultati se dele po tome da li metoda koristi duboko učenje ili ne. Posmatrajući tabelu, konvoluciona neuronska mreža je ostvarila najveću tačnost kao metoda sa dubokim učenjem, a XGBoost daje najbolje rezultate među metodama koje ne koriste duboko učenje. Najmanje rezultate daje SVM sa polinomijalnim kernelom.
+
+U FSDD bazi dato je 10 labela, pa klasifikator radi odličan posao da pretpostavi u koju kategoriju labela određeni zvuk spada (cifra od 0 do 9).
+
+U ovoj bazi podataka, određene reči mogu lako da se pomešaju na spektrogramu, pa su neke vrednosti vrlo blizu odlučnoj granici i da pomute labele. Iz tog razloga, rezultati SVM metode su veoma dobri.
+
+Konvoluciona neuronska mreža je metoda koja je najviše razrađena u ovom projektu. Deep learning metode povoljnije su za feature extraction proces, koji je neophodan kako bismo sa spektrograma mogli lepo da izvučemo informacije o zvuku. Cross entropy loss, to jest log loss odlično funkcioniše kao speech recognition loss funkcija pošto ljudsko uho reaguje logaritamski. To znači da je naše uho daleko osetljivije na niske frekvencije, primećujući razliku od svega nekoliko herca pri frekvencijama od ~200Hz, dok je ta razlika potpuno neprimetna na frekvencijama od nekoliko kHz. Osetljivost je pri dnu približno linearna, dok sa porastom frekvencije postaje logaritamska.
+
+Gledajući ova dva faktora u obzir, očekivano je da će performansa CNN-a biti najbolja.
+
+Rezultati koji su odađeni na srpskoj bazi podataka dosta su slabiji u poređenju sa engleskom bazom. Srpska baza pravljena je u amaterskim uslovima: mikrofon slabijeg kvaliteta, dosta šuma se može čuti u samim snimcima, nisu svi zvuci iste jačine, kao ni dužine. Ovi faktori dosta utiču na kvalitet spektrograma, na kome ima dosta više šuma u poređenju sa spektrogramom engleske baze.
+
+Rezultate vizuelno možemo prikazati matricama konfuzije. 
+
+![Rezultati](static/images/LinearSVM.png)
+
+![Rezultati](static/images/LogisticRegression.png)
+
+![Rezultati](static/images/RandomForest.png)
+
+![Rezultati](static/images/XGB.png)
+### Zaključak
+
+Projekat koristi FSDD bazu podataka za poređenje performansi pri prepoznavanju govora između sledećih metoda: SVM, MFCCs, CNN, Random Forest, XGBoost i logistička regresija. Uz ovu i samostalno napravljenu srpsku bazu podataka, ove metode su se pokazale kao veoma uspešne pri detektovanju izgovorenih reči. CNN model je imao najveću uspešnost pri prevođenju reči, a SVM sa RBF kernelom najmanju. Tačnost između metoda varira od 51.45% do 97.28%, pa je zaključak ovog rada da je tačnost CNN modela značajno veća od ostalih testiranih modela.
```

Comment:
1. Prva recenica zakljucka treba da kaze sta je projekat, a posle recite kakva baza postoji
2. Iskreno nikog ne zanima najmanja postignuta tacnost, jer mogu ja dodjem kazem moja metoda lupanja random daje 10% tacnosti (mos' misliti), tako da to mozete obrisati
3. Nije zakljucak rada da je CNN bolji, to pise u tabeli. Zakljucak je sta mozete zakljuciti iz toga i npr kakve to implikacije moze da ima ka daljem radu